lib/rules/no-useless-escape.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @fileoverview Look for useless escapes in strings and regexes
+ * @author Onur Temizkan
+ * @copyright 2016 Onur Temizkan. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+var VALID_STRING_ESCAPES = {
+    "\\": true,
+    "n": true,
+    "r": true,
+    "v": true,
+    "t": true,
+    "b": true,
+    "f": true,
+    "u": true,
+    "x": true
+};
+
+var VALID_REGEX_ESCAPES = {
+    "\\": true,
+    ".": true,
+    "-": true,
+    "^": true,
+    "$": true,
+    "*": true,
+    "+": true,
+    "?": true,
+    "{": true,
+    "}": true,
+    "[": true,
+    "]": true,
+    "|": true,
+    "(": true,
+    ")": true,
+    "b": true,
+    "c": true,
+    "d": true,
+    "D": true,
+    "f": true,
+    "n": true,
+    "r": true,
+    "s": true,
+    "S": true,
+    "t": true,
+    "v": true,
+    "w": true,
+    "W": true,
+    "x": true,
+    "u": true
+};
+
+var nodeEscapes, nodeQuote;
```

Comment:
Why are these variables declared way out here? Move them into the check function.